From 1cbbc03c4b37d57760c57bd2e0b0f89d744a5795 Mon Sep 17 00:00:00 2001
From: Stefan Dirsch <sndirsch@suse.de>
Date: Tue, 7 Oct 2014 11:32:01 +0200
Subject: vmmouse_client.h: Include xorg-server.h before xf86_OSproc.h

xf86_OSproc.h cannot be included without first including xorg-server.h.

Without this the build fails on systems with the latest glibc,
throwing this error:

In file included from /usr/include/string.h:634:0,
                 from /usr/include/xorg/os.h:53,
                 from /usr/include/xorg/misc.h:115,
                 from /usr/include/xorg/window.h:50,
                 from /usr/include/xorg/globals.h:7,
                 from /usr/include/xorg/opaque.h:34,
                 from /usr/include/xorg/xf86_OSproc.h:127,
                 from vmmouse_client.h:42,
                 from vmmouse_client.c:38:
/usr/include/xorg/os.h:579:1: error: expected identifier or '(' before '__extension__'
strndup(const char *str, size_t n);

This is caused by HAVE_STRNDUP not being set (it is set from xorg-server.h),
causing os.h to redefine it.

Signed-off-by: Stefan Dirsch <sndirsch@suse.de>
Reviewed-by: Daniel Stone <daniels@collabora.com>
Acked-by: Thomas Hellstrom <thellstrom@vmware.com>

diff --git a/shared/vmmouse_client.h b/shared/vmmouse_client.h
index b01bf08..1de51e5 100644
--- a/shared/vmmouse_client.h
+++ b/shared/vmmouse_client.h
@@ -39,6 +39,7 @@
 #ifndef _VMMOUSE_CLIENT_H_
 #define _VMMOUSE_CLIENT_H_
 
+#include "xorg-server.h"
 #include "xf86_OSproc.h"
 
 /*
-- 
cgit v0.10.2

